libagent/ssh/__init__.py

```diff
@@ -241,13 +275,16 @@ def _dummy_context():
 
 def _get_sock_path(args):
     sock_path = args.sock_path
-    if not sock_path:
-        if args.foreground:
-            log.error('running in foreground mode requires specifying UNIX socket path')
-            sys.exit(1)
-        else:
-            sock_path = tempfile.mktemp(prefix='trezor-ssh-agent-')
-    return sock_path
+    if sock_path:
+        return sock_path
+    elif args.foreground:
+        log.error('running in foreground mode requires specifying %s', SOCK_TYPE_PATH)
+        sys.exit(1)
+    elif sys.platform == 'win32':
+        suffix = random.choices(string.ascii_letters, k=10)
```

Comment:
`mkstemp` is not applicable because it will try to create and open the file, which can't be done for a pipe address. `mktemp` will also try to check for the file's existence, which may or may not work for a pipe address, and is deprecated.